servant/src/Servant/Links.hs

```diff
@@ -649,7 +649,7 @@ simpleToLink _ toA _ = toLink toA (Proxy :: Proxy sub)
 -- Erroring instance for 'HasLink' when a combinator is not fully applied
 instance TypeError (PartialApplication HasLink arr) => HasLink ((arr :: a -> b) :> sub)
   where
-    type MkLink (arr :> sub) _ = TypeError (PartialApplication HasLink arr)
+    type MkLink (arr :> sub) _ = TypeError (PartialApplication (HasLink :: * -> Constraint) arr)
```

Comment:
This was added 2 weeks ago: https://github.com/haskell-servant/servant/commit/42b7d0eb9bc36e02fc4aa2656bcbde9aa540aaee

GHC 9.2 is stricter about this:

> GHC is stricter about checking for out-of-scope type variables on the right-hand sides of associated type family instances that are not bound on the left-hand side.

[source](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/9.2.1-notes.html)

Reply:
aab7e0d added another instance of this problem: https://github.com/haskell-servant/servant/blob/7a770b5a1e0fc250c5f19c0629329e28b69151c0/servant-server/src/Servant/Server/Internal.hs#L824
`HasServer` needs to become `(HasServer :: * -> [*] -> Constraint)` (and `Constraint` needs to be added to the import list).